packages/web/src/hooks/swap/data/use-swap.tsx

```diff
@@ -154,19 +154,23 @@ export const useSwap = ({ tokenA, tokenB, direction, slippage, swapFee = 15 }: U
     }
 
     const amount = estimatedSwapResult.amount;
+
     return direction === "EXACT_IN"
       ? makeDisplayTokenAmount(tokenB, amount)?.toString() || null
       : makeDisplayTokenAmount(tokenA, amount)?.toString() || null;
   }, [debouncedSwapAmount, error, swapState, estimatedSwapResult, isTyping]);
 
+  const swapFeeRate = useMemo(() => (swapFee || 0) / 100, [swapFee]);
+
   const tokenAmountLimit = useMemo(() => {
     if (estimatedAmount && !Number.isNaN(slippage)) {
+      const amountWithSwapFee = BigNumber(estimatedAmount.toString()).multipliedBy(1 - swapFeeRate / 100);
       const tokenAmountLimit =
         direction === "EXACT_IN"
-          ? BigNumber(estimatedAmount)
+          ? BigNumber(amountWithSwapFee)
               .multipliedBy((100 - slippage) / 100)
               .toNumber()
-          : BigNumber(estimatedAmount)
+          : BigNumber(amountWithSwapFee)
               .multipliedBy((100 + slippage) / 100)
               .toNumber();
```

Comment:
`toNumber` may lead to precision loss. If precision loss is not important, it doesn't matter, but using `toString` can prevent this problem.

Also, Creating a `BigNumber` object and then creating another `BigNumber` from `tokenAmountLimit` is unnecessary.

maybe like this?

```ts
const tokenAmountLimit = BigNumber(estimatedAmount.toString())
  .multipliedBy(1 - swapFeeRate / 100)
  .multipliedBy(direction === "EXACT_IN" ? (100 - slippage) / 100 : (100 + slippage) / 100)
  .toString();
```